parser/Builder.cc

```diff
@@ -176,10 +176,14 @@ class Builder::Impl {
         return cond;
     }
 
+    // Record a diagnostic error, and set internal state that will trigger a YYERROR on the next DIAGCHECK. Use this
+    // function if there's no reasonable tree that can be returned, as it will trigger the parser to throw away
+    // arbitrary amounts of the input tree, potentially leading to LSP taking the slow path unnecessarily.
```

Comment:
```suggestion
    // Record a diagnostic error, and set internal state that will emit an `error` token on the next DIAGCHECK.
    // Use this function if there's no reasonable tree that can be returned, as it will trigger the parser to
    // throw away arbitrary amounts of the input tree, potentially leading to LSP taking the slow path unnecessarily.
```